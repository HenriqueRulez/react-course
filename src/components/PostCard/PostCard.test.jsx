import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import postCardPropsMock from "./mock";

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props}/>);

    expect(screen.getByRole('img', { title: props.title })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: props.title })).toBeInTheDocument();
    expect(screen.getByText('Post body')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props}/>);

    //Avoid accessing direct child node
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot()
  });
});