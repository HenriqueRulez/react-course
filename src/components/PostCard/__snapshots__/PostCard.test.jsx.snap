// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Post title"
    src="img/img.png"
  />
  <section
    class="post-content"
  >
    <h2>
      Post title
    </h2>
    <p>
      Post body
    </p>
  </section>
</div>
`;
